loader/src/main/kotlin/net/weavemc/loader/bootstrap/transformer/SafeTransformer.kt

```diff
@@ -19,9 +23,19 @@ internal interface SafeTransformer : ClassFileTransformer {
         protectionDomain: ProtectionDomain?,
         classfileBuffer: ByteArray
     ) = runCatching {
-        transform(loader, className, classfileBuffer)
+        val bytes = transform(loader, className, classfileBuffer)
+        if (checkBytecode && bytes != null) {
+            ClassNode().also {
+                ClassReader(bytes).accept(it, ClassReader.EXPAND_FRAMES)
```

Comment:
- Why do we need a ClassNode? seems wasteful to allocate and parse etc. for each transformed class, a classreader parses all the information we need without parsing the rest of the class file
- Why do we need to expand frames? seems wasteful again, we don't seem to need expanded frames

Reply:
This was supposed to be off by default, my bad